src/Atmos/Parameterizations/CloudPhysics/Microphysics.jl

```diff
@@ -1,271 +1,575 @@
-        _grav::FT = grav(param_set)
-        _ρ_cloud_liq::FT = ρ_cloud_liq(param_set)
-        _MP_n_0::FT = MP_n_0(param_set)
+    λ::FT = FT(0)
 
-        lambda::FT = (FT(8) * π * _ρ_cloud_liq * _MP_n_0 / ρ / q_rai)^FT(1 / 4)
-        rain_w = gamma_9_2 * v_c / FT(6) * sqrt(_grav / lambda)
+    if q > FT(0)
+        λ = (α * n0 * gamma(β + FT(1)) / ρ / q)^FT(1/(β + 1))
     end
+    return λ
+end
 
-    return rain_w
+"""
+    unpack_params(param_set, microphysics_param_set, ρ, q_)
+
+ - `param_set` - abstract set with earth parameters
+ - `microphysics_param_set` - abstract set with microphysics parameters
+ - `q_` - specific humidity
+ - `ρ` - air density
+
+Utility function that unpacks microphysics parameters.
+"""
+function unpack_params(param_set::APS, ice_param_set::AIPS, ρ::FT,
+                       q_ice::FT) where {FT<:Real}
+    #TODO - make ρ and q_ice optional
+    _n0::FT = n0(ice_param_set)
+    _α::FT  = α(param_set, ice_param_set)
+    _β::FT  = β(ice_param_set)
+
+    return (_n0, _α, _β)
+end
+function unpack_params(param_set::APS, rain_param_set::ARPS, ρ::FT,
+                       q_rai::FT) where {FT<:Real}
+    #TODO - make q_rai optional
+    _n0::FT = n0(rain_param_set)
+    _α::FT  = α(param_set, rain_param_set)
+    _β::FT  = β(rain_param_set)
+    _γ::FT  = γ(rain_param_set)
+    _δ::FT  = δ(rain_param_set)
+    _ζ::FT  = ζ_rai(param_set, ρ)
+    _η::FT  = η(rain_param_set)
+
+    return (_n0, _α, _β, _γ, _δ, _ζ, _η)
+end
+function unpack_params(param_set::APS, snow_param_set::ASPS, ρ::FT,
+                       q_sno::FT) where {FT<:Real}
+
+    _n0::FT = n0_sno(snow_param_set, q_sno, ρ)
+    _α::FT  = α(snow_param_set)
+    _β::FT  = β(snow_param_set)
+    _γ::FT  = γ(snow_param_set)
+    _δ::FT  = δ(snow_param_set)
+    _ζ::FT  = ζ(snow_param_set)
+    _η::FT  = η(snow_param_set)
+
+    return (_n0, _α, _β, _γ, _δ, _ζ, _η)
 end
 
 """
-    conv_q_vap_to_q_liq(q_sat, q)
+    supersaturation(param_set, q, ρ, T, Liquid())
+    supersaturation(param_set, q, ρ, T, Ice())
 
-where:
-- `q_sat` - PhasePartition at equilibrium
-- `q`     - current PhasePartition
+ - `param_set` - abstract set with earth parameters
+ - `q` - phase partition
+ - `ρ` - air density,
+ - `T` - air temperature
+ - `Liquid()`, `Ice()` - liquid or ice phase to dispatch over.
 
-Returns the q_liq tendency due to condensation/evaporation.
-The tendency is obtained assuming a relaxation to equilibrium with
-constant timescale.
-"""
-function conv_q_vap_to_q_liq(
-    param_set::APS,
-    q_sat::PhasePartition{FT},
-    q::PhasePartition{FT},
-) where {FT <: Real}
-    _τ_cond_evap::FT = τ_cond_evap(param_set)
-    return (q_sat.liq - q.liq) / _τ_cond_evap
+Returns supersaturation (qv/qv_sat -1) over water or ice.
+"""
+function supersaturation(param_set::APS, q::PhasePartition{FT}, ρ::FT, T::FT,
+                         ::Liquid) where {FT<:Real}
+
+    q_sat::FT = q_vap_saturation_generic(param_set, T, ρ)
```

Comment:
```suggestion
    q_sat::FT = q_vap_saturation(param_set, T, ρ)
```
(with updated argument list for the phase partitioning). If you use this function, this makes it possible to consider supersaturation over a mixture of liquid and ice.

Reply:
@tapios  - Right, but I'm using supersaturation here to compute evaporation of rain or deposition, sublimation of snow specifically. In those cases I should be looking at either liquid only or ice only conditions?

Reply:
Ok if it's only for that purpose.